src/darwin/frida-helper-backend-glue.m

```diff
@@ -1450,6 +1466,74 @@ static void frida_darwin_helper_backend_launch_using_sbs (NSString * identifier,
   return killed_pid;
 }
 
+static gboolean
+frida_has_active_prewarm (gint pid)
+{
+  gboolean result = false;
+  int mib_argmax[] = { CTL_KERN, KERN_ARGMAX };
+  int mib_args[] = { CTL_KERN, KERN_PROCARGS2, 0 };
+  size_t size;
+  gint32 arg_max = 0, argc;
+  guint8 * buffer = NULL, * cursor, * end;
+
+  size = sizeof (arg_max);
+
+  if (sysctl (mib_argmax, G_N_ELEMENTS (mib_argmax), &arg_max, &size, NULL, 0) != 0)
+    goto beach;
+  if (arg_max < sizeof (argc))
+    goto beach;
+
+  buffer = g_malloc (arg_max);
+  if (buffer == NULL)
+    goto beach;
+
+  mib_args[2] = pid;
+  size = arg_max;
+
+  if (sysctl (mib_args, G_N_ELEMENTS (mib_args), buffer, &size, NULL, 0) != 0)
+    goto beach;
+  if (size < sizeof (argc))
+    goto beach;
+
+  argc = *(gint32 *)buffer;
+  end = buffer + size;
+  cursor = buffer + sizeof (argc);
+
+  /* Skip executable name */
+  cursor = frida_skip_string (cursor, end);
+
+  /* Skip args */
+  while (argc-- != 0)
+    cursor = frida_skip_string (cursor, end);
+
+  /* Iterate environment */
+  while (cursor < end)
+  {
+    if (strstr ((char *)cursor, "ActivePrewarm=1"))
+    {
+      result = true;
+      break;
+    }
+    cursor = frida_skip_string (cursor, end);
+  }
+
+beach:
+  g_free (buffer);
+
+  return result;
+}
+
+static guint8 *
+frida_skip_string (guint8 * cursor, guint8 * end)
+{
+  while (cursor < end && *cursor != 0)
+    cursor++;
+  while (cursor < end && *cursor == 0)
+    cursor++;
```

Comment:
Can there be more than one NUL that we want to skip? Wondering if this loop is needed and we could instead skip the one NUL we're already on. (That made the previous loop terminate.)